expression/builtin_control.go

```diff
@@ -553,11 +553,11 @@ func (c *ifFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 	if err != nil {
 		return nil, err
 	}
-	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, evalTps, types.ETInt, evalTps, evalTps)
+
+	bf, err := newBaseBuiltinFuncWithFieldTypes(ctx, c.funcName, args, evalTps, args[0].GetType().Clone(), retTp.Clone(), retTp.Clone())
```

Comment:
`args[0].GetType().Clone()` should be field type of `int/bool`?

Reply:
`args[0].GetType().Clone()` must be bool type because `WrapForIsTrue` has been called previously